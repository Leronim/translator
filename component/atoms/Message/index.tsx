import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

const message = [
    'Реакт',
    'Нейтив',
    'иногда',
    'доставляет',
    'проблем',
    'моей',
    'заднице',
    'зато',
    'весело'
]

const Message = () => {
    return(
        <View style={style.container}>
            {message.map((item: string, index: number) => {
                return(
                    <View key={index} style={style.text}>
                        <Text>{item} </Text>
                    </View>
                )
            })}
        </View>
    )
}

const style = StyleSheet.create({
    container: {
        borderWidth: 2,
        borderRadius: 40,
        borderColor: '#252525',
        padding: 17,
        flexDirection: 'row',
        flexWrap: 'wrap',
        width: 250,
        justifyContent: 'center'
    },
    text: {
        marginRight: 10,
    }
})

export default Message;