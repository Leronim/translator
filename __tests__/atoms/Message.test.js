import React from 'react';
import Message from '../../component/atoms/Message';
import { render } from '@testing-library/react-native';

test('Message render', () => {
    const tree = render(<Message/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('Message view', () => {
    const { getByTestId } = render(<Message/>);
    
    expect(getByTestId('view').props.children).toHaveLength(8);
})