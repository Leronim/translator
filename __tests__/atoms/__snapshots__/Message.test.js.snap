// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message render 1`] = `
<View
  style={
    Object {
      "borderColor": "#252525",
      "borderRadius": 40,
      "borderWidth": 2,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "padding": 17,
      "width": 250,
    }
  }
  testID="view"
>
  <View
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    <Text>
      Реакт
       
    </Text>
  </View>
  <View
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    <Text>
      Нейтив
       
    </Text>
  </View>
  <View
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    <Text>
      доставляет
       
    </Text>
  </View>
  <View
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    <Text>
      проблем
       
    </Text>
  </View>
  <View
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    <Text>
      моей
       
    </Text>
  </View>
  <View
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    <Text>
      заднице
       
    </Text>
  </View>
  <View
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    <Text>
      зато
       
    </Text>
  </View>
  <View
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    <Text>
      весело
       
    </Text>
  </View>
</View>
`;
